src/sparseml/transformers/finetune/data/custom.py

```diff
@@ -55,9 +58,16 @@ def get_raw_dataset(self, *_ignore, **__ignore) -> Union[DatasetDict, Dataset]:
             raw_dataset = super().get_raw_dataset()
 
         if self.preprocessing_func is not None:
+            func = (
+                self.preprocessing_func
+                if isinstance(self.preprocessing_func, Callable)
+                else PreprocessingFunctionRegistry.get_value_from_registry(
```

Comment:
besides registry, let's also enable support for a custom pathway like `"path/to/file.py:function_name"`

see https://github.com/neuralmagic/deepsparse/blob/4e791ee9843f87dd35a3c026bb7b63f5e20d6aef/src/deepsparse/legacy/loggers/build_logger.py#L69